### DISCUSSION TITLE
Way to display emojis in thread.messages content

### DISCUSSION THREAD

**Discussion by lokize** [discordjs/discord.js#9386] [Q&A]
Hi, all my msgs is returning with numbers and expressions for emojis, is there any way to display emojis in message like replace all emoticons code and display images?

![image](https://user-images.githubusercontent.com/19701630/232041012-75fb29b3-7a77-42d1-a4fd-ab79985105da.png)

![image](https://user-images.githubusercontent.com/19701630/232041063-6ae330bf-d39a-4d36-b4ce-a9306f595f1a.png)

I managed to get the emoticons by id with the method:

client.emojis.cache.get("id)

however, I believe there must be another easier way to get the emojis without having to replace all the codes that appear in the message content...

**Comment by sdanialraza** [comment]
Once you get the emoji from the cache, stringifying it would show in the message, you can either use template literals or use [GuildEmoji.toString()](https://old.discordjs.dev/#/docs/discord.js/main/class/GuildEmoji?scrollTo=toString) method.

**Reply by lokize** [reply]
i made with worked but i think its not the best way yet, here is my code:

```
const client = new Client({
     intents: [
     GatewayIntentBits.Guilds,
     GatewayIntentBits.GuildMessages,
     GatewayIntentBits.MessageContent,
     GatewayIntentBits.GuildPresences,
      ],
      });
       client.login('my bot token');
      const channel = await client.channels.fetch('chanell id');
      const thread = await channel.threads.fetch('thread id');
      const content= await thread.messages.fetch({ });
      content.forEach((message) => {
      message.content = message.content.replace(/<:[a-zA-Z0-9_]+:[0-9]+>/g,
        (match) => {
          const id = match.replace(/<:[a-zA-Z0-9_]+:/g, "").replace(/>/g, "");
         
          let myemoji = client.emojis.cache.find((emoji) => emoji.id === id).toJSON();

         myemoji = `<img src="${myemoji.url}" alt="${myemoji.name}" width=40 height=40 />`;

         return myemoji;
        }
      );
    });

```

**Reply by sdanialraza** [reply]
As you have `Guilds` intent, you don't need to fetch the channel, just get it from the cache as all channels are cached when you have the `Guilds` intent.

**Reply by lokize** [reply]
thanks for help!
